src/tarball.py

```diff
@@ -270,10 +271,14 @@ def _run(self, sub_id):
                 tarball_name
             )
             tarball_url = self._push_tarball(tarball_path)
-            commit_tags, commit_message, branch_tip = self._get_commit_info(
-                self._service_config.kdir, commitid, config_tree, config_branch)
-            self._update_node(checkout_node, describe, version, tarball_url,
-                              commit_tags, commit_message, branch_tip)
+            try:
+                commit_tags, commit_message, branch_tip = self._get_commit_info(
+                    self._service_config.kdir, commitid, config_tree, config_branch)
+                self._update_node(checkout_node, describe, version, tarball_url,
+                                  commit_tags, commit_message, branch_tip)
+            except subprocess.CalledProcessError as err:
+                self._logger.error(f"Git commit info retrieval failed: {err}")
+                self._update_failed_checkout_node(checkout_node, "git-commit-error", str(err))
```

Comment:
Please use `_` in place of `-` to match the pattern with other error codes.
```suggestion
                self._update_failed_checkout_node(checkout_node, "git_commit_error", str(err))
```

Reply:
It might be broken tree, or commit got removed before we finished retrieving tree. We can check if it repeats too often, but for now we just should expect it as glitch on remote side.